### PR TITLE
Clarify API concepts

### PR DESCRIPTION
**Pull request by alexandratran**
Combine API concepts into one page and update title to make clear that it is explanatory/intro content. Add links to reference content prominently at the top of the page.

Preview: https://docs.metamask.io/fix-concepts/wallet/concepts/apis/

### CODE REVIEW

File: wallet/concepts/apis.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+:::note
+All RPC method requests can return errors.
+Make sure to handle errors for every call to
+[`window.ethereum.request(args)`](../reference/provider-api.md#windowethereumrequestargs).
+:::
+
+### Restricted methods
+
+MetaMask introduced web3 wallet permissions in [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255).
```
Comment (BelfordZ) [inline wallet/concepts/apis.md:74]: IMO we should avoid using the term 'web3'. Maybe im wrong there, but afaik it's become somewhat meaningless. 
```suggestion
MetaMask introduced wallet permissions as specified by [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255).
```

**Review by BelfordZ — APPROVED** [review]
this is fantastic. Great improvement to the docs!

**Review by jiexi — APPROVED** [review]

**Review by vandan — APPROVED** [review]
Agreed. These are excellent improvements.

**Review by bgravenorst — APPROVED** [review]
Yep, I think this will avoid any confusion.

**Pull request merged**